src/core/Akka.Remote/Transport/Helios/HeliosTcpTransport.cs

```diff
@@ -281,32 +281,36 @@ public HeliosTcpTransport(ActorSystem system, Config config)
         /// <param name="remoteAddress">TBD</param>
         /// <exception cref="HeliosConnectionException">TBD</exception>
         /// <returns>TBD</returns>
-        protected override Task<AssociationHandle> AssociateInternal(Address remoteAddress)
+        protected override async Task<AssociationHandle> AssociateInternal(Address remoteAddress)
```

Comment:
This is the key change here. I don't think we need to throw some Helios exceptions here, cause no one is waiting them and we are not following the interface - if we know what is happening, we should throw **InvalidAssociationException**, if not - like usual - just don't catch it.